gradle-guide-error-prone/src/main/java/com/palantir/gradle/guide/errorprone/NonAbstractGradleType.java

```diff
@@ -53,18 +58,48 @@ public final class NonAbstractGradleType extends GradleGuideBugChecker
             .named("create");
     private static final Supplier<Type> CLASS_TYPE_SUPPLIER = Suppliers.typeFromString("java.lang.Class");
 
+    private static final Set<String> SUPPORTED_PROPERTY_TYPES = Set.of(
+            "org.gradle.api.provider.Property",
+            "org.gradle.api.provider.ListProperty",
+            "org.gradle.api.provider.SetProperty",
+            "org.gradle.api.provider.MapProperty",
+            "org.gradle.api.file.RegularFileProperty",
+            "org.gradle.api.file.DirectoryProperty",
+            "org.gradle.api.provider.Provider",
+            "org.gradle.api.model.ObjectFactory");
+
     @Override
     public Description matchClass(ClassTree tree, VisitorState state) {
         if (tree.getKind().equals(Kind.INTERFACE)) {
             return Description.NO_MATCH;
         }
 
-        if (tree.getModifiers().getFlags().contains(Modifier.ABSTRACT)) {
-            return Description.NO_MATCH;
-        }
-
         if (IS_TASK.matches(tree, state)) {
-            return buildDescription(tree).build();
+            if (!tree.getModifiers().getFlags().contains(Modifier.ABSTRACT)) {
+                return buildDescription(tree)
+                        .setMessage("Gradle managed Task types must be abstract classes.")
```

Comment:
This error message is not very descriptive. See the message in the `@BugPattern` this was before - it really should have plenty of information. I think for this one, we should just not set a custom message so it uses the default.